src/berry_mill/builder.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import annotations
+import argparse
+import logging
+
+from typing import List
+from typing_extensions import Unpack
+from lxml import etree
+import os
+import sys
+import shutil
+import tempfile
+
+from urllib.parse import urlparse, quote
+from typing import Dict
+from berry_mill.kiwiapp import KiwiAppLocal, KiwiAppBox
+from kiwi.kiwi import KiwiError
+from platform import machine
+from berry_mill.kiwrap import KiwiParent
+
+from berry_mill.params import KiwiBuildParams, KiwiParams
+
+log = logging.getLogger('kiwi')
+
+class KiwiBuilder(KiwiParent):
+    """
+    Main Class for Berrymill to prepare the kiwi-ng system (box)build calls
+    """
+    # TODO handle TypeError
+    def __init__(self, descr:str, **kw: Unpack[KiwiBuildParams]):
+        super().__init__(descr=descr,
+                        profile=kw.get("profile"),
+                        debug=kw.get("debug"))
+        
+        self._params:Dict[KiwiBuildParams] = kw
+
+        if "target_dir" in self._params:
```

Comment:
And that's exactly the opposite: you need to know if `target_dir` is not empty. Because it very much can be `""` or `None` as well, so this will crash or work wrong. To fix this:

```python
if self._params.get("target_dir"):
```

:-)

Reply:
done